pyicloud/services/findmyfriends.py

```diff
@@ -0,0 +1,84 @@
+"""Find my Friends service."""
+from __future__ import absolute_import
+import json
+
+
+class FindFriendsService(object):
+    """
+    The 'Find my Friends' iCloud service
+
+    This connects to iCloud and returns friend data including the near-realtime
+    latitude and longitude.
+    """
+
+    def __init__(self, service_root, session, params):
+        self.session = session
+        self.params = params
+        self._service_root = service_root
+        self._friend_endpoint = "%s/fmipservice/client/fmfWeb/initClient" % (
+            self._service_root,
+        )
+        self.response = {}
+
+    def refresh_data(self):
+        """
+        Refreshes all data from Find my Friends endpoint,
+
+        This ensures that the location data is up-to-date.
+
+        """
+        params = dict(self.params)
+        # This is a request payload we mock to fetch the data
+        mock_payload = json.dumps(
+            {
+                "clientContext": {
+                    "appVersion": "1.0",
+                    "contextApp": "com.icloud.web.fmf",
+                    "mapkitAvailable": True,
+                    "productType": "fmfWeb",
+                    "tileServer": "Apple",
+                    "userInactivityTimeInMS": 537,
+                    "windowInFocus": False,
+                    "windowVisible": True,
+                },
+                "dataContext": None,
+                "serverContext": None,
+            }
+        )
+        req = self.session.post(self._friend_endpoint, data=mock_payload, params=params)
+        # Update the response for normal execution flow
+        self.response = req.json()
+        # FEAT: Return a value to support monkey-patching
+        return self.response
```

Comment:
Do not return, and remove 2 up comments.

Don't know exactly monkey-patching but we are not using it for now.

Maybe later on with another test rework.

```suggestion

        self.response = req.json()
```

Reply:
fine.

Reply:
Why not, but not in this PR because you should modify all services, they all work about the same

Reply:
I am willing to unify even more services with an abstract class